### DISCUSSION TITLE
Can't set Kitty as default terminal

### DISCUSSION THREAD

**Discussion by rocktimsaikia** [kovidgoyal/kitty#5599] [Q&A] [answered]
I have installed kitty through `apt` and it's located in `/usr/bin/kitty`. Then I set it as the default terminal with `update-alternatives`. But even after setting that the gnome-terminal still pops up when I open the term with the keyboard shortcut (`ctrl+alt+t`) instead of Kitty.

OS: Ubuntu 22.04.1 LTS x86_64
Kernel: 5.19.0-051900-generic

![image](https://user-images.githubusercontent.com/33410545/196512622-34d8f112-f5f5-4346-87a3-1f0739cd7772.png)

**Comment by page-down** [comment, marked as answer]
You need to make sure that `x-terminal-emulator` is configured as the default terminal in the GNOME Desktop settings for the **current user**.
Or just configure it as `kitty`.

Changing only the `alternatives` of `x-terminal-emulator` (global, for all local users) won't ensure that `CTRL+ALT+T` opens the terminal you want.

It is also recommended to use the latest official release. Version 0.21 in Ubuntu is outdated and is not supported here.

**Reply by rocktimsaikia** [reply]
Thanks, that makes sense. But I was able to do it by changing the value in the **gsettings** itself 
```sh
gsettings set org.gnome.desktop.default-applications.terminal exec 'kitty'
```
Also, I have switched to the latest official binary now that you have mentioned it.

**Comment by juanMarinero** [comment]
I think `$(which kitty)` is preferable than raw `'kitty'`

```
sudo update-alternatives --install /usr/bin/x-terminal-emulator x-terminal-emulator $(which kitty) 50
# sudo update-alternatives --config x-terminal-emulator # config (interactive)
update-alternatives --display x-terminal-emulator # check

gsettings set org.gnome.desktop.default-applications.terminal exec $(which kitty)
gsettings get org.gnome.desktop.default-applications.terminal exec # check
```

Btw. Next to open scripts with Vim in a Kitty terminal (Ubuntu 24.04) no matter which is the default terminal:

```
#!/bin/bash

file01="/usr/local/share/applications/vim.desktop"

# Check if the file exists
if [ ! -f "$file01" ]; then
    echo "Error: $file01 does not exist."
    exit 1
fi

# Make a backup of the original file
sudo cp "$file01" "${file01}.bak"

# Get the actual home directory
home_dir=$(echo $HOME)

# Use sed to make the changes (comment original 3 lines, add direct after new 3 lines)
sudo sed -i "
    /^TryExec=vim/s/^/#/
    /^Exec=vim %F/s/^/#/
    /^Terminal=true/s/^/#/
    /^#Terminal=true/a\\
TryExec=${home_dir}/.local/kitty.app/bin/kitty\\
Exec=${home_dir}/.local/kitty.app/bin/kitty vim %F\\
Terminal=false
" "$file01"

# Check the changes
echo "Checking changes:"
grep -E "TryExec=|Exec=|Terminal=" "$file01"

# plain text files to be opened with Vim
xdg-mime default vim.desktop text/plain

# Update the desktop database after making these changes
sudo update-desktop-database /usr/local/share/applications
```

**Comment by chrnz008** [comment]
u can use update alternative to set kitty as default terminal but it is limited ive tried it but i cant open .desktop exec with kitty
